improver/threshold.py

```diff
@@ -211,14 +218,14 @@ def __init__(
                 "Can only collapse over a realization coordinate or a percentile "
                 "coordinate that has been rebadged as a realization coordinate."
             )
-        else:
-            self.collapse_coord = collapse_coord
+        self.collapse_coord = collapse_coord
 
         self.vicinity = None
         if vicinity is not None:
             if isinstance(vicinity, numbers.Number):
                 vicinity = [vicinity]
-            self.vicinity = [float(x) for x in vicinity]
+            if vicinity:  # To handle an empty list being provided.
+                self.vicinity = [float(x) for x in vicinity]
```

Comment:
I'm fairly sure this can now be simplified to this:
```
if vicinity:
    if iterable(vicinity):
        self.vicinity = [float(x) for x in vicinity]
    else:
        self.vicinity = [vicinity]
```

Reply:
I'm afraid I have no idea what the `iterable` function you've proposed here is. Could you point me to the module that contains this as it would be very useful to use in this way without having to use `isinstance`.

Reply:
I've simplified the if statement so far.